Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/async/AsyncMessageProcessHelper.java

```diff
@@ -385,17 +389,24 @@ private Blob getBlobFromMCCIIN000002UV01(final MCCIIN000002UV01 ack) {
             baOutStrm.close();
             marshaller.marshal(oJaxbElement, baOutStrm);
             final byte[] buffer = baOutStrm.toByteArray();
-            asyncMessage = getSession().getLobHelper().createBlob(buffer);
-            getSession().close();
+
+            session = getSession();
+            asyncMessage = session.getLobHelper().createBlob(buffer);
         } catch (final Exception e) {
             LOG.error("Exception during Blob conversion :" + e.getMessage(), e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+            hibernateUtil.closeSessionFactory();
```

Comment:
The documentation recommends we should close sessionFactory once the application exit